Source/Core/Core.js

```diff
@@ -301,13 +306,24 @@ Number.extend('random', function(min, max){
 	return Math.floor(Math.random() * (max - min + 1) + min);
 });
 
-// forEach, each
+// forEach, each, keys
 
-var hasOwnProperty = Object.prototype.hasOwnProperty;
-Object.extend('forEach', function(object, fn, bind){
-	for (var key in object){
-		if (hasOwnProperty.call(object, key)) fn.call(bind, object[key], key, object);
+Object.extend({
+
+	keys: function(object){
+		var keys = [];
+		eachKey(object, function(key, value){
+			if (hasOwnProperty.call(object, key)) keys.push(key);
+		}, this);
+		return keys;
+	},
+
+	forEach: function(object, fn, bind){
+		Object.keys(object).forEach(function(key){
```

Comment:
Array.forEach is not defined at this point yet.. (but only a few lines after this). Maybe do the Object.* methods after the Array ones?